src/isar_turtlebot/robotinterface.py

```diff
@@ -123,12 +134,39 @@ def log_status(
     def get_inspection_references(
         self, vendor_mission_id: Any, current_step: Step
     ) -> Sequence[Inspection]:
-        return []
+        now: datetime = datetime.utcnow()
+
+        if isinstance(current_step, TakeImage):
+            self.bridge.visual_inspection.register_run_id(run_id=vendor_mission_id)
+            pose: Pose = self.get_robot_pose()
+            image_metadata: ImageMetadata = ImageMetadata(
+                start_time=now,
+                time_indexed_pose=TimeIndexedPose(pose=pose, time=now),
+                file_type=config.get("metadata", "image_filetype"),
+            )
+            image_ref: ImageReference = ImageReference(
+                id=vendor_mission_id, metadata=image_metadata
+            )
+
+        return [image_ref]
 
     def download_inspection_result(
         self, inspection: Inspection
     ) -> Optional[InspectionResult]:
-        return None
+        if isinstance(inspection, ImageReference):
```

Comment:
Should be support for thermal image as well

Reply:
Will remove support for `ThermalImageStep`, so this can be handled in a separate PR.